api/v1beta1/machine_types.go

```diff
@@ -125,6 +125,31 @@ type MachineSpec struct {
 	// +optional
 	FailureDomain *string `json:"failureDomain,omitempty"`
 
+	// The minimum number of seconds for which a Machine should be ready before considering it available.
+	// Defaults to 0 (Machine will be considered available as soon as the Machine is ready)
+	// NOTE: this field will be considered only for computing v1beta2 conditions.
+	// +optional
+	// TODO: This field will be added in the v1beta2 API, and act as a replacement of existing MinReadySeconds in
+	//  MachineDeployment, MachineSet and MachinePool
+	// MinReadySeconds int32 `json:"minReadySeconds,omitempty"`
+
+	// readinessGates specifies additional conditions to include when evaluating Machine readiness.
+	// A Machine is ready when `BootstrapConfigReady`, `InfrastructureReady`, `NodeHealthy` and `HealthCheckSucceeded` (if present) are "True";
+	// if other conditions are defined in this field, those conditions should be "True" as well for the Machine to be ready.
+	//
+	// This field can be used e.g.
+	// - By Cluster API control plane providers to extend the semantic of the Ready condition for the Machine they
+	//   control, like the kubeadm control provider adding ReadinessGates for the APIServerPodHealthy, SchedulerPodHealthy conditions, etc.
+	// - By external controllers, e.g. responsible to install special software/hardware on the Machines
+	//   to include the status of those components into ReadinessGates (by surfacing new conditions on Machines and
+	//   adding them to ReadinessGates).
```

Comment:
Listing with bullets here won't render nicely, please re-format to sentences

Reply:
Fixed